gse-app/src/main/java/com/powsybl/gse/app/ProjectPane.java

```diff
@@ -203,6 +216,119 @@ protected void updateItem(Object value, boolean empty) {
                     setGraphic(null);
                 } else {
                     fillCellInfosForObject(value);
+                    if (value == null) {
+                        GseUtil.setWaitingText(this);
+                    } else {
+                        if (value instanceof String) {
+                            setText((String) value);
+                            setGraphic(getTreeItem().getGraphic());
+                            setTextFill(Color.BLACK);
+                            setOpacity(1);
+                        } else if (value instanceof ProjectNode) {
+                            ProjectNode node = (ProjectNode) value;
+                            setText(node.getName());
+                            setGraphic(getTreeItem().getGraphic());
+                            setTextFill(Color.BLACK);
+                            setOpacity(node instanceof UnknownProjectFile ? 0.5 : 1);
+
+                            setOnDragDetected(event -> {
```

Comment:
This callback is reset at each refresh of the treeview. This is a design (and performance) issue.

Reply:
//TreeItem Documentation
TreeItem is not a Node, and therefore no visual events will be fired on the TreeItem. To get these events, it is necessary to add relevant observers to the TreeCell instances (via a custom cell factory )